### PR TITLE
Refresh Dep Viewer

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Adding a button for user to refresh dep viewer manually.

RefreshDepViewer

Tooltip:
image

HoverOverState:
RefreshDepViewerHoverOverState

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@DynamoDS/dynamo 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCoreWpf/Views/Core/NodeView.xaml [hunk 1]
```diff
@@ -5,7 +5,6 @@
              xmlns:dynui="clr-namespace:Dynamo.UI.Controls"
              xmlns:p="clr-namespace:Dynamo.Wpf.Properties"
              xmlns:configuration="clr-namespace:Dynamo.Configuration;assembly=DynamoCore"
-             xmlns:fa="http://schemas.fontawesome.io/icons/"
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Views/Core/NodeView.xaml]: might this negatively effect nodeViews which reference font awesome?

Comment (QilongTang) [reply src/DynamoCoreWpf/Views/Core/NodeView.xaml]: Shouldn't be the case, at least this name space is not referenced at all.. I can also remove if there are known issues but I can't think of any side effect

File: src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml [hunk 2]
```diff
@@ -120,7 +128,7 @@
                 </DataGridTextColumn>
                 <!-- Package Version -->
                 <DataGridTextColumn 
-                    Header="Version" 
+                    Header="{x:Static w:Resources.VersionHeaderText}"
```
Comment (mjkkirschner) [inline src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml]: thanks

File: src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml.cs [hunk 3]
```diff
@@ -246,6 +246,11 @@ public void Dispose()
             loadedParams.CurrentWorkspaceCleared -= OnWorkspaceCleared;
             WorkspaceModel.DummyNodesReloaded -= TriggerDependencyRegen;
         }
+
+        private void Refresh_MouseLeftButtonDown(object sender, System.Windows.Input.MouseButtonEventArgs e)
+        {
+            DependencyRegen(currentWorkspace);
```
Comment (mjkkirschner) [inline src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml.cs]: does this function have testing?

File: src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml (hunk 2 above)
Comment (QilongTang) [reply src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml]: Thanks, this should fix the localization bug

File: src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml.cs (hunk 3 above)
Comment (QilongTang) [reply src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml.cs]: Needs UI automation test for this one.

File: src/WorkspaceDependencyViewExtension/Properties/Resources.resx [hunk 4]
```diff
@@ -359,13 +359,22 @@
...
+    <value>Package</value>
+  </data>
   <data name="ProvideFeedbackButton" xml:space="preserve">
     <value>Provide Feedback</value>
   </data>
   <data name="ProvideFeedbackError" xml:space="preserve">
     <value>Could not redirect to the Dynamo forum page for feedback:</value>
   </data>
+  <data name="RefreshButtonTooltipText" xml:space="preserve">
+    <value>Click to refresh package dependencies.</value>
```
Comment (mjkkirschner) [inline src/WorkspaceDependencyViewExtension/Properties/Resources.resx]: what do you think of 
`Click to recalculate workspace dependencies` - I know today we only show package dependencies so not sure if we should make this more general for the future.

Also refresh to me - is a bit ambiguous between the view of the dependencies or the actual dependencies on disk or the download state etc. - what do you think?

File: src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml [hunk 5]
```diff
@@ -96,17 +97,24 @@
...
-                    IsReadOnly="True" 
-                    Width="26">
+                    IsReadOnly="True"
+                    Width="34">
+                    <DataGridTemplateColumn.Header>
+                        <fa:ImageAwesome Name="Refresh" 
+                                         Icon="Refresh" 
+                                         Foreground="{DynamicResource MemberButtonText}" 
+                                         ToolTip="{x:Static w:Resources.RefreshButtonTooltipText}" 
+                                         MouseLeftButtonDown="Refresh_MouseLeftButtonDown"/>
```
[the remark this answers is not in the document]
Comment (mjkkirschner) [reply src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml]: thanks, any thoughts on the hover interaction?

**Comment by QilongTang** [comment]
I tried a bunch of ways to add hover over state, this way seems more consistent with what we have with the buttons for download. However I did notice that the buttons in dep viewer are not really consistent with what we have defined in Dynamo. Bummer.

**Comment by mjkkirschner** [comment]
@QilongTang LGTM

**Pull request merged**